### ISSUE TITLE
Formatting error erases code on file save

### ISSUE THREAD

**Issue by Sorrien** [dbaumgarten/yodk#14]
I just ran into an issue where attempting to save with what the plugin considered improper code had the plugin attempting to overwrite my current code with what seems like what it last considered to be valid code. I was able to undo the change, but a plugin I have to fight is a plugin I won't use.

**Comment by dbaumgarten** [comment]
Hi,
well this is something that definetly should never happen.
Could you please provide a sample of code that lead to this behaviour?
This would be really helpful in fixing this issue.

**Comment by Sorrien** [comment]
Firstly, thank you for the quick response.

So I pasted in 
`message="hello world!"`
saved the file, everything's fine.
replace all the previous code with
`7 % 2`
and then when I try to save it overwrites back to
`message="hello world!"`

If I don't replace all of the code then I just get an error in the console
ex:
`message="hello world!"
7 % 2`
Error:
`[Error - 3:39:22 PM] Request textDocument/formatting failed.
  Message: Parser error at Line: 2, Coloumn: 1: Expected a statement. Found Token: '7'(Number)


  Code: 0`

I do not see this error when my code is replaced, only when it leaves it as I wrote it.

**Comment by dbaumgarten** [comment]
I was able to reproduce the error and found out what was causing it.

It seems like I misunderstood the language server protocol specification.
When vscode request the server to format a file, it just passes the URI if the file to format. Th server then loaded the file from the disk, computed formatting operations and sent them back to vscode.
The problem here is, at this point in time the content of the file on the disc was outdated. Even if it was a format-on-save, the file would only be saved AFTER the formatting.
Because of this, the formatter always was "one version behind" with the file's contents, which lead to really strange behavior.

I changed the language server, to keep track of the current content of the file using the documentChanged-events and formatting now happens in the version of the file received in the last documentChanged-event.


Regarding the error you got for ```message="hello world!" 7 % 2```: The error is correct, this is no valid yolol. The "7%2" is an expression. It can not exist alone, as it would not do anything. ```message="hello world!" number=7 % 2``` would be correct.

However, printing an ugly error message when the user tries to format invalid code it not really nice.Therefore I disabled the error message. The red lines in the code should be enough to indicate errors.

Tl;Dr: I did something stupid,  release v0.0.7 should now correctly format things.
Upload and verification of the new version is currently happening. The update should be available in vscode in a few minutes.

**Comment by Sorrien** [comment]
Good work. I agree that the code is incorrect but that the error message is a clunky user experience. I will check out the new version when it's available.

Thanks for making this project!

**Comment by Sorrien** [comment]
I just tested using the same steps with version 0.0.8 and the issue no longer occurs
